### PR TITLE
ci: Fix and build Starter Projects

### PR DESCRIPTION
**Pull request by dplewis**
### New Pull Request Checklist

- [ ] I am not disclosing a [vulnerability](https://github.com/parse-community/Parse-SDK-iOS-OSX/security/policy).
- [ ] I am creating this PR in reference to an [issue](https://github.com/parse-community/Parse-SDK-iOS-OSX/issues?q=is%3Aissue).

### Issue Description

The code for building starter projects does exists but was never used and didn't work when used.
Because they weren't use a lot of issues could have been caught.

### Approach

* Use SPM for all projects
* Update Rakefile with new commands
* Add build to CI
* Remove package release build

### TODOs before merging

- [x] Add tests
- [x] Add changes to documentation (guides, repository pages, in-code descriptions)

### TODOs after merging
- [ ] Fix watchOS starter project compilation errors.

### CODE REVIEW

**Comment by dplewis** [comment]
@mtrezza This is ready for review

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -30,6 +30,10 @@ For analyzing bugs, creating bug fixes and features we recommend to clone this r
    ```
    bundle exec rake test:ios
    ```
+5. Ensure the project builds. You can use the [Starter Projects](https://github.com/dplewis/Parse-SDK-iOS-OSX/tree/master/ParseStarterProject) to perform integration tests.
```
Comment (mtrezza) [inline CONTRIBUTING.md:33]:
```suggestion
5. Ensure the project builds. You can use the [Starter Projects](https://github.com/parse-community/Parse-SDK-iOS-OSX/tree/master/ParseStarterProject) to perform integration tests.
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -24,9 +23,9 @@ jobs:
           - test:twitter_utils:ios
           - test:parseui:all
           - test:parse_live_query:all
-          - package:release
```
Comment (mtrezza) [inline .github/workflows/ci.yml:27]: This removes `package:release`, is that for building the assets?

**Review by mtrezza — COMMENTED** [review]
Almost...

File: .github/workflows/ci.yml (hunk 2 above)
Comment (dplewis) [reply .github/workflows/ci.yml:27]: Yes

Comment (mtrezza) [reply .github/workflows/ci.yml:27]: I'll add this as a note manually to the next release changelog.

**Review by mtrezza — APPROVED** [review]
Looks good; CI passes before docs change, so I'll just merge.

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [2.7.2](https://github.com/parse-community/Parse-SDK-iOS-OSX/releases/tag/2.7.2)

**Pull request merged**